mgmake/generate.go

```diff
@@ -18,77 +21,142 @@ type templateTargets struct {
 	Args       []string
 }
 
-// GenerateMakefile is a mage target that ...
-func GenerateMakefile(makefile string) error {
-	const defaultTargets = "mage_targets"
-	genDir := filepath.Dir(makefile)
-	mgDefaultTargets := fmt.Sprintf("%s.mk", filepath.Join(genDir, strcase.ToKebab(defaultTargets)))
-	if makefile == mgDefaultTargets {
-		return fmt.Errorf("%s has the same name as the default %s makefile", makefile, mgDefaultTargets)
+// nolint: gochecknoglobals
+var (
+	makefiles = make(map[string]makefile)
+)
+
+type makefile struct {
+	Path          string
+	DefaultTarget string
+}
+
+type Makefile struct {
+	Namespace     interface{}
+	Path          string
+	DefaultTarget func()
+}
+
+func GenerateMakefiles(mks ...Makefile) {
+	for _, i := range mks {
+		if i.Path == "" {
+			panic("Path needs to be defined")
+		}
+		namespace := defaultNamespace
+		if i.Namespace != nil {
+			namespace = reflect.TypeOf(i.Namespace).Name()
+		}
+		defaultTarget := runtime.FuncForPC(reflect.ValueOf(i.DefaultTarget).Pointer()).Name()
+		defaultTarget = strings.TrimPrefix(defaultTarget, "main.")
+		defaultTarget = strings.TrimPrefix(defaultTarget, namespace+".")
+		defaultTarget = strings.Split(defaultTarget, "-")[0]
```

Comment:
@odsod thoughts on this? The reason behind it is because the function names are originally `main.All` & `main.Proto.Hej-fm` 
Would like a better solution for extracting the function name but this is what i came up with

Reply:
As long as it supports also `Proto.Hej` and it throws an error if the parsing turns up something unexpected, then let's go for it!

Reply:
Sounds good, added verification to make sure that the end result only contains letters